src/IslandoraUtils.php

```diff
@@ -768,6 +768,28 @@ protected function getParentsByEntityReference(ContentEntityInterface $entity, a
     return $parents;
   }
 
+  /**
+    * Insert a value or key/value pair after a specific key in an array.  If key doesn't exist, value is appended
+    * to the end of the array.
+    *
+    * This is not islandora specific, but a useful function that neither PHP nor Drupal provide.
+    * (is there still hope after 16 years?  https://www.drupal.org/project/drupal/issues/66183 )
+    *
+    * @param array $array
+    * @param string $key
+    * @param array $new
+    *
+    * @return array
+    *
+    * Credit: https://gist.github.com/wpscholar/0deadce1bbfa4adb4e4c
+    */
+  function array_insert_after( array $array, $key, array $new ) {
+	  $keys = array_keys( $array );
+	  $index = array_search( $key, $keys );
+	  $pos = false === $index ? count( $array ) : $index + 1;
+	  return array_merge( array_slice( $array, 0, $pos ), $new, array_slice( $array, $pos ) );
+  }
+  
```

Comment:
Fair number of things in this chunk, from: https://github.com/Islandora/islandora/actions/runs/10498083873/job/29082307172?pr=904

```
 FILE: /home/runner/work/islandora/islandora/build_dir/src/IslandoraUtils.php
--------------------------------------------------------------------------------
FOUND 47 ERRORS AND 3 WARNINGS AFFECTING 20 LINES
--------------------------------------------------------------------------------
 772 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 772 | WARNING | [ ] Line exceeds 80 characters; contains 113 characters
 773 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 773 | ERROR   | [ ] Doc comment short description must be on a single line,
     |         |     further text should be a separate paragraph
 774 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 775 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 775 | WARNING | [ ] Line exceeds 80 characters; contains 96 characters
 776 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 776 | WARNING | [ ] Line exceeds 80 characters; contains 96 characters
 777 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 778 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 778 | ERROR   | [ ] Missing parameter comment
 779 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 779 | ERROR   | [ ] Missing parameter comment
 780 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 780 | ERROR   | [ ] Missing parameter comment
 781 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 782 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 783 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 784 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 784 | ERROR   | [x] Return comment indentation must be 3 spaces, found 1
     |         |     spaces
 785 | ERROR   | [x] Expected 3 space(s) before asterisk; 4 found
 786 | ERROR   | [x] Visibility must be declared on method "array_insert_after"
 786 | ERROR   | [ ] Method name "IslandoraUtils::array_insert_after" is not in
     |         |     lowerCamel format
 786 | ERROR   | [x] Expected 0 spaces after opening parenthesis; 1 found
 786 | ERROR   | [x] There should be no white space after an opening "("
 786 | ERROR   | [x] There should be no white space before a closing ")"
 786 | ERROR   | [x] Expected 0 spaces before closing parenthesis; 1 found
 787 | ERROR   | [x] Spaces must be used to indent lines; tabs are not allowed
 787 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 3
 787 | ERROR   | [x] There should be no white space after an opening "("
 787 | ERROR   | [x] There should be no white space before a closing ")"
 788 | ERROR   | [x] Spaces must be used to indent lines; tabs are not allowed
 788 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 3
 788 | ERROR   | [x] There should be no white space after an opening "("
 788 | ERROR   | [x] There should be no white space before a closing ")"
 789 | ERROR   | [x] Spaces must be used to indent lines; tabs are not allowed
 789 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 3
 789 | ERROR   | [x] TRUE, FALSE and NULL must be uppercase; expected "FALSE"
     |         |     but found "false"
 789 | ERROR   | [x] There should be no white space after an opening "("
 789 | ERROR   | [x] There should be no white space before a closing ")"
 790 | ERROR   | [x] Spaces must be used to indent lines; tabs are not allowed
 790 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 3
 790 | ERROR   | [x] There should be no white space after an opening "("
 790 | ERROR   | [x] There should be no white space after an opening "("
 790 | ERROR   | [x] There should be no white space before a closing ")"
 790 | ERROR   | [x] There should be no white space after an opening "("
 790 | ERROR   | [x] There should be no white space before a closing ")"
 790 | ERROR   | [x] There should be no white space before a closing ")"
 792 | ERROR   | [x] Whitespace found at end of line
```